### ISSUE TITLE
[Bug]: (v7) Cannot prerender resource routes

### ISSUE THREAD

**Issue by k0michi** [remix-run/react-router#12196]
### What version of React Router are you using?

7.0.0-pre.1, 7.0.0-pre.2

### Steps to Reproduce

1. Go to https://stackblitz.com/edit/github-6rufwv
2. Run `pnpm build`

### Expected Behavior

It should build successfully.

### Actual Behavior

It fails with the following message.

Note that node version is irrelevant to this error; I got the same error with node 22.10.0.

```
~/projects/github-6rufwv
❯ pnpm build
 WARN  Unsupported engine: wanted: {"node":">=20.0.0"} (current: {"node":"v18.20.3","pnpm":"8.15.6"})

> @ build /home/projects/github-6rufwv
> react-router build

vite v5.4.10 building for production...
✓ 41 modules transformed.
Generated an empty chunk: "resource".
build/client/.vite/manifest.json                1.57 kB │ gzip:  0.42 kB
build/client/assets/root-DQHep_n7.css           6.79 kB │ gzip:  2.02 kB
build/client/assets/resource-l0sNRNKZ.js        0.00 kB │ gzip:  0.02 kB
build/client/assets/with-props-CMSwQhCY.js      0.19 kB │ gzip:  0.16 kB
build/client/assets/root-DxNDofi5.js            0.78 kB │ gzip:  0.45 kB
build/client/assets/home-DtVDxzf_.js            3.74 kB │ gzip:  1.67 kB
build/client/assets/index-DAl0xit8.js          99.52 kB │ gzip: 33.13 kB
build/client/assets/entry.client-Do1hcxZM.js  137.13 kB │ gzip: 44.36 kB
✓ built in 1.64s
vite v5.4.10 building SSR bundle for production...
✓ 11 modules transformed.
build/server/.vite/manifest.json                0.23 kB
build/server/assets/server-build-DQHep_n7.css   6.79 kB
build/server/index.js                          10.54 kB
Prerender: Generated build/client/resource.data
The following error is a bug in Remix; please open an issue! https://github.com/remix-run/remix/issues/new
Error: 
    at invariant (file:///home/projects/github-6rufwv/node_modules/.pnpm/react-router@7.0.0-pre.1_react-dom@18.3.1_react@18.3.1/node_modules/lib/server-runtime/headers.ts:27:9)
    at handleResourceRequest (file:///home/projects/github-6rufwv/node_modules/.pnpm/react-router@7.0.0-pre.1_react-dom@18.3.1_react@18.3.1/node_modules/lib/server-runtime/sessions.ts:271:49)
x Build failed in 556ms
[react-router] Prerender: Received a 500 status code from `entry.server.tsx` while prerendering the `/resource/` path.
Unexpected Server Error
    at validatePrerenderedResponse (/home/projects/github-6rufwv/node_modules/.pnpm/@react-router+dev@7.0.0-pre.1_@react-router+serve@7.0.0-pre.1_react-router@7.0.0-pre.1_typescript@5.6.3_vite@5.4.10/node_modules/@react-router/dev/dist/vite/plugin.js:1252:11)
    at prerenderRoute (/home/projects/github-6rufwv/node_modules/.pnpm/@react-router+dev@7.0.0-pre.1_@react-router+serve@7.0.0-pre.1_react-router@7.0.0-pre.1_typescript@5.6.3_vite@5.4.10/node_modules/@react-router/dev/dist/vite/plugin.js:1230:30) {
  code: 'PLUGIN_ERROR',
  plugin: 'react-router',
  hook: 'writeBundle'
}
 ELIFECYCLE  Command failed with exit code 1.
```

**Comment by k0michi** [comment]
The same error can be seen in #12181.

**Comment by brophdawg11** [comment]
This should be resolved in `7.0.0-pre.3` if you want to give it a shot 👍
